src/test/32-pipelines.ts

```diff
@@ -0,0 +1,126 @@
+import { expect } from "chai";
+import { Fabric } from "../jsC8";
+
+describe("Manipulating pipelines", function() {
+    this.timeout(50000);
+  
+    let fabric: Fabric;
+    const testUrl: string =
+      process.env.TEST_C8_URL || "https://qa2.eng1.macrometa.io";
+  
+    before(async () => {
+      fabric = new Fabric({
+        url: testUrl,
+        c8Version: Number(process.env.C8_VERSION || 30400)
+      });
+      await fabric.login("guest", "root", "guest");
+      fabric.useTenant("guest");
+      fabric.useFabric("_system");
+          
+    });
+  
+    after(() => {
+      fabric.close();
+    });
+  
+    describe("fabric.getAllPipelines", () => {
+      it("returns an empty array or an array of all pipelines", async() => {
+        const regions = ["qa2-asia-east1"];
+        const isEnabled = false;
+        const config = {
+          "input": {
+          "type": "c8db",
+          "c8db": {
+            "name": "collection_name1"
+          }
+        },
+          "output": {
+          "type": "c8streams",
+          "c8streams": {
+            "name": "stream_name1",
+            "local": true
+          }
+        }
+        };
+        const allPipelines = await fabric.getPipelines();
+        const pipelinesCount = allPipelines.length;
+        expect(allPipelines).to.be.an('array');
+        const pipeline = await fabric.pipeline('pipeline1');
+        await pipeline.create(regions,isEnabled, config);
+        const updatedAllPipelines = await fabric.getPipelines();
+        const expectedPipelinesCount = pipelinesCount + 1;
+        expect(updatedAllPipelines.length).to.equal(expectedPipelinesCount);
+      });
+    });
+  
+    describe("pipeline.create", () => {
+        it("creates a new pipeline", async () => {
+            const regions = ["qa2-asia-east1"];
+            const isEnabled = false;
+            const config = {
+                "input": {
+                    "type": "c8db",
+                    "c8db": {
+                        "name": "collection_name2"
+                    }
+                    },
+                "output": {
+                    "type": "c8streams",
+                    "c8streams": {
+                        "name": "stream_name2",
+                        "local": true
+                    }
+                }
+            };
+            const pipeline = await fabric.pipeline('pipeline2');
+            const response = await pipeline.create(regions,isEnabled, config);
+            expect(response).to.equal('Stream Created Successfully');
```

Comment:
Rather than checking for a string it will be better if we check for the success response code.

Reply:
Backend is not sending that. For POST AND PUT, they are sending these strings.